safe/gui/tools/people_in_buildings_dialog.py

```diff
@@ -0,0 +1,380 @@
+        geometry_feature = feature.geometry()
+        geometry_layer = layer.geometry()
+        centroid = geometry_feature.centroid()
+        if not centroid.within(geometry_layer):
+            return 0
+        return geometry_feature.area()
+
+    @staticmethod
+    def _get_levels(attributes, levels_attribute):
+        """Try to extract the levels from the attributes dict as an int.
+        ..Note: If the levels of this building has not been set we will
+        assume it to be 1.
+
+        :param attributes: The attributes dict from the feature
+        :type attributes: dict
+
+        :param levels_attribute: The name of the column that contains layer
+        information
+        :type levels_attribute: basestring
+
+        :returns: Number of levels (default 1)
+        :rtype: int
+        """
+        levels = attributes[levels_attribute]
+        # get the number of levels as a integer
+        if not levels:
+            return 1
+        try:
+            # get the number of levels as an intiger
+            return int(levels)
+        except ValueError:
+            return 1
+
+    @staticmethod
+    def _get_residential_proportion(attributes, building_use):
+        """The proportion of the surface to be used for residential.
+
+        :param attributes: The feature's attribute dictionary lookup.
+        :type attributes: dict
+
+        :param building_use: The name of the attribute containing the
+            building use.
+        :type building_use: basestring
+
+        :returns: The residential proprtion
+        :rtype: int, float
+
+        ...Note: The residential proportion is based on the building type.
+        Currently the types 'Residential' and 'House' are considered to be
+        used 100% for residential.
+        """
+        use = attributes[building_use]
+        if use in ['Residential', 'residential', 'house', 'House']:
+            return 1
+        # if use == '' or use is None or isinstance(use, QPyNullVariant):
+        #     # Default to residential, if it is unclassified
+        #     return 1
+        return 0
+
+    @staticmethod
+    def _feature_fully_in_extent(layer, feature):
+        """Determine wether the feature is fully within the layer's extent.
+
+        :param layer: The layer to be considered.
+        :type layer: VectorLayer
+
+        :param feature: The feature to be considered.
+        :type feature:
+
+        :return: The result of the test.
+        :rtype: bool
+
+        ...Note: This operation uses the internal centroid.
+
+        """
+        extent = layer.extent()
+        geometry_extent = QgsGeometry.fromRect(extent)
+        geometry_feature = feature.geometry()
+        intersection = geometry_feature.intersection(geometry_extent)
+        if intersection.area() < geometry_feature.area():
+            return False
+        return True
+
+    def estimate_people_in_buildings(self):
+        building_layer_name = self.buildingLayerComboBox.currentText()
```

Comment:
docstring - would be good to have a fairly detailed description here of what the underlying logic is.